### PR TITLE
Allow generation without clone of easybuild-framework and move location of source

### LINKED ISSUE #88: support building docs without API part
With the changes in #79, having `easybuild-framework` cloned into `src` is a hard requirement for the docs build to work:

```
$ ls src
ls: src: No such file or directory
$ make preview
mkdocs serve
INFO     -  Building documentation...
INFO     -  Cleaning site directory
FileNotFoundError: [Errno 2] No such file or directory: 'py3.py' -> 'src/easybuild/tools/py2vs3/py2.py'
make: *** [preview] Error 1
```

That's because the `gen_ref_pages.py` script blindly assumes that `src` contains a copy of the `easybuild-framework` repository.

We should allow building the documentation without the API part, to speed up local development of the documentation.

While doing that, we should also use `src/easybuild-framework` instead of just `src`.

### PR DESCRIPTION
**Pull request by branfosj**
fixes #88

### CODE REVIEW

File: docs/gen_ref_pages.py [hunk 1]
```diff
@@ -8,17 +8,22 @@
 import mkdocs_gen_files
 
 
+if not Path('src/easybuild-framework/easybuild').is_dir():
```
Comment (boegel) [inline docs/gen_ref_pages.py:11]: Maybe checking for `easybuild/framework` is even better?

File: docs/gen_ref_pages.py [hunk 2]
```diff
@@ -8,17 +8,22 @@
...
+    print("WARNING  -  EasyBuild Framework is not available. API documentation will not be generated.")
+    exit(0)
+
+
 # remove the problematic py2.py and symlink the py3.py in its place
-Path("src/easybuild/tools/py2vs3/py2.py").unlink(missing_ok=True)
-Path("src/easybuild/tools/py2vs3/py2.py").symlink_to(Path("py3.py"))
+Path("src/easybuild-framework/easybuild/tools/py2vs3/py2.py").unlink(missing_ok=True)
+Path("src/easybuild-framework/easybuild/tools/py2vs3/py2.py").symlink_to(Path("py3.py"))
```
Comment (boegel) [inline docs/gen_ref_pages.py:18]: Avoid repeating the "src/easybuild-framework", introduce a constant on top of the script?

**Review by boegel — CHANGES_REQUESTED** [review]

File: docs/gen_ref_pages.py (hunk 2 above)
Comment (branfosj) [reply docs/gen_ref_pages.py]: done in ee556fc

File: docs/gen_ref_pages.py (hunk 1 above)
Comment (branfosj) [reply docs/gen_ref_pages.py]: I've made a further change in ee556fc that means that this check is sufficient for the rest of the script to function. If `src/easybuild-framework/easybuild` has nothing in it then we'll generate no API documentation.

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**